temoa/extensions/framework.py

```
@@ -0,0 +1,237 @@
+    """Fail if disabled extensions with strict guards own tables populated with data."""
+    enabled_ids = {spec.extension_id for spec in enabled_specs}
+    for spec in get_known_extension_specs().values():
+        if spec.extension_id in enabled_ids:
+            continue
+        if not spec.fail_if_tables_populated_when_disabled:
+            continue
+
+        populated: list[str] = []
+        for table in spec.owned_tables:
+            if _table_has_rows(con, table):
+                populated.append(table)
```

Ruff (0.15.21)

[warning] Use a list comprehension to create a transformed list

(PERF401)